.github/workflows/ci.yml

```diff
@@ -0,0 +1,51 @@
+# For more information see: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow
+
+name: CI
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  lint-and-types:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [23.x]
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'npm'
+    - run: npm ci
+    - run: npm run lint
+    - run: npm run test:types
+
+  tests:
+    strategy:
+      fail-fast: false
+      matrix:
+        node:
+          - version: 23.x
+          - version: 22.x
+          - version: 20.x
+        os:
+          - macos-latest
+          - ubuntu-latest
+          - windows-latest
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Node.js ${{ matrix.node.version }}
+      uses: actions/setup-node@v4
+      with:
+        node-version: ${{ matrix.node.version }}
+        cache: 'npm'
+    - run: npm ci
+    - run: npm run test:unit
+    - run: npm run test:e2e
```

Comment:
Should these be switched to `node --run …`?

Reply:
maybe